src/test/kotlin/org/rust/cargo/commands/RustfmtTest.kt

```diff
@@ -310,4 +300,25 @@ class RustfmtTest : RsWithToolchainTestBase() {
 
         action.actionPerformed(e)
     }
+
+    /**
+     * Note: [expectedTextSupplier] is called right before [action].
+     */
+    private fun doTest(
+        tree: FileTree,
```

Comment:
You can pass `FileTreeBuilder.() -> Unit` instead of `FileTree`. It allows you to git rid of `fileTree` in every call of `doTest `